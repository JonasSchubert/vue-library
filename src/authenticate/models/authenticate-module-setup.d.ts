export interface AuthenticateModuleSetup {
  cookieKeyAuthenticationToken: string;
  daysTilExpiredAuthenticationCookie: number;
}
