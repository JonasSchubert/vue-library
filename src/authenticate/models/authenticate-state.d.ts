import { LoginResponse } from "./login-response";

export interface AuthenticateState<T extends LoginResponse> {
  data?: T;
  error?: Error;
  isLoading: boolean;
}
