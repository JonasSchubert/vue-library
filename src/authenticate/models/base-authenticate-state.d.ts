import { AuthenticateState } from "./authenticate-state";
import { LoginResponse } from "./login-response";

export interface BaseAuthenticateState {
  authenticate: AuthenticateState<LoginResponse>;
  requests: number;
}
