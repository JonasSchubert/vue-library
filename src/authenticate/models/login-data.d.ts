export interface LoginData {
  password: string;
  saveLoginDataTemporary: boolean;
  userName: string;
}
