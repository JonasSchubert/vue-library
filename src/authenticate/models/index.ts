export * from './authenticate-module-setup.d';
export * from './authenticate-state.d';
export * from './base-authenticate-state.d';
export * from './login-data.d';
export * from './login-response.d';
export * from './validate-token-data.d';
