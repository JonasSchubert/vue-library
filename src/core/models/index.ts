export * from './base-state.d';
export * from './entity.d';
