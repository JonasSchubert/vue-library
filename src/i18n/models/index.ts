export * from './action-load-payload.d';
export * from './i18n-module-config.d';
export * from './i18n-state.d';
