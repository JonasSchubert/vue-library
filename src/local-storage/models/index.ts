export * from './local-storage-history.d';
export * from './local-storage-state.d';
export * from './local-storage-settings.d';
