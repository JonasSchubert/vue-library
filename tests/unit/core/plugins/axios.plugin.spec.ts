import { AxiosStatic } from 'axios';
import { Store } from 'vuex';
import { BaseState } from '@/core/models';
import { createAxiosPlugin } from '@/core/plugins/axios.plugin';

describe('plugins/axios', () => {
  const axiosMock = {
    defaults: {
      baseURL: undefined
    },
    interceptors: {
      request: { use: jest.fn() },
      response: { use: jest.fn() }
    }
  };
  const storeMock = { commit: jest.fn() };
  const VueMock = { $http: undefined };

  beforeEach(() => {
    axiosMock.interceptors.request.use.mockReset();
    axiosMock.interceptors.response.use.mockReset();
    storeMock.commit.mockReset();
  });

  test('should set Vue.$http, baseUrl', () => {
    // Arrange
    const baseUrl = "http://localhost:15151/vue-library/api/v1/";

    // Act
    createAxiosPlugin(baseUrl).install(VueMock, {
      axios: (axiosMock as unknown as AxiosStatic),
      store: (storeMock as unknown as Store<BaseState>)
    });

    // Assert
    expect(axiosMock.defaults.baseURL).toBe('http://localhost:15151/vue-library/api/v1/');
    expect(VueMock.$http).toMatchSnapshot();
  });

  test.todo('interceptors should have an unit test');
});
