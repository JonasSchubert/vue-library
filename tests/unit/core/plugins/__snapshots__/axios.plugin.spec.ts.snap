// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/axios should set Vue.$http, baseUrl 1`] = `
Object {
  "defaults": Object {
    "baseURL": "http://localhost:15151/vue-library/api/v1/",
  },
  "interceptors": Object {
    "request": Object {
      "use": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
    "response": Object {
      "use": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    },
  },
}
`;
